eclipse-theia/theia
# [electron] [plugin] Use macaddress instead of getmac

Signed-off-by: Rob Moran <rob.moran@arm.com>

#### What it does
This PR fixes https://github.com/eclipse-theia/theia/issues/6384 by swapping out the `getmac` dependency for `macaddress`.

`getmac` and one of it's dependencies rely on incompatible versions of the same package (`editions`) which are incorrectly squashed together when built with `electron-builder` due to [this issue](https://github.com/electron-userland/electron-builder/issues/3185).

The swapped package avoids this issue and also seems more popular on [npm](https://www.npmjs.com/package/macaddress).

As the package is MIT and has no dependencies, it satisfied the [simplified process](https://github.com/eclipse-theia/theia/wiki/Registering-CQs#wip---new-ecd-theia-intellectual-property-clearance-approach-experimental) for introducing dependencies to Theia.

#### How to test
Please use the plugin host both in the browser and electron environments and confirm the `macaddress` call [returns a unique ID](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/plugin/node/env-node-ext.ts#L37) which persists across sessions.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
